src/_igraph/graphobject.c

```diff
@@ -7813,6 +7813,65 @@ PyObject *igraphmodule_Graph_minimum_cycle_basis(
   return result_o;
 }
 
+
+PyObject *igraphmodule_Graph_simple_cycles(
+  igraphmodule_GraphObject *self, PyObject *args, PyObject *kwds
+) {
+  PyObject *mode_o = Py_None;
+  PyObject *output_o = Py_None;
+  PyObject *min_cycle_length_o = Py_None;
+  PyObject *max_cycle_length_o = Py_None;
+
+  // argument defaults: no cycle limits
+  igraph_integer_t mode = IGRAPH_OUT;
+  igraph_integer_t min_cycle_length = -1;
+  igraph_integer_t max_cycle_length = -1;
+  igraph_bool_t use_edges = false;
+
+  static char *kwlist[] = { "mode", "min", "max", "output", NULL };
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "|OOOO", kwlist, &mode_o, &min_cycle_length_o, &max_cycle_length_o, &output_o))
+    return NULL;
+
+  if (mode_o != Py_None && igraphmodule_PyObject_to_integer_t(mode_o, &mode))
+    return NULL;
+
+  if (min_cycle_length_o != Py_None && igraphmodule_PyObject_to_integer_t(min_cycle_length_o, &min_cycle_length))
+    return NULL;
+
+  if (max_cycle_length_o != Py_None && igraphmodule_PyObject_to_integer_t(max_cycle_length_o, &max_cycle_length))
+    return NULL;
+
+  if (igraphmodule_PyObject_to_vpath_or_epath(output_o, &use_edges))
+    return NULL;
+
+  igraph_vector_int_list_t vertices;
+  igraph_vector_int_list_init(&vertices, 0);
```

Comment:
Even though we are allocating a vector of size 0 here, the allocation may still fail (we never really allocate a buffer of zero bytes, there's a minimum amount of memory that we allocate even for an empty vector). Therefore, this call needs to be guarded against failures with the usual pattern:

```
if (igraph_vector_int_list_init(&vertices, 0)) {
    igraphmodule_handle_igraph_error();
    return NULL;
}
```

Same for the allocation of the edge vector below.

Reply:
Thanks, good point, I introduced these two guards in my latest commits